Add infix forms for backtick functions using auxiliary objects

hvpaiva opened the pull request:
## Description

This pull request adds infix forms for some backtick functions that are normally unary relations by using auxiliary objects. The main motivation for this change is to improve readability and fluency when using backtick functions in an infix manner.

Examples:

```kotlin
"   " `should be` Blank
"" `should be` Empty
```

is more readable and fluent than:

```kotlin
"   ".`should be blank`()
"".`should be empty`()
```

I have made these changes wherever it was possible to replace the previous unary relation with an object representing it. In some cases, the improvement in readability might be less significant, but using specialized methods, in my opinion, is still preferable to using `shouldBeEqualsTo()` everywhere.

```kotlin
list `should be` Empty
```

is more readable and fluent than:

```kotlin
list.`should be empty`()
```

but not as much as, even I prefer with `Empty`:

```kotlin
list `should be equals to` emptyList<Int>()
```

### Regarding the implementation:

- I added @Suppress annotations in some cases where the parameter argument is just a marker and not really used. 
- For the case of `file 'should be' aFile`, I used a non-conventional name to avoid shadowing the class one. So I suppress it too.


### Concerning the unit tests:

- I'm not entirely sure if I've followed the project's unit testing conventions. I made an effort to adhere to the existing conventions, but please let me know if any changes are needed to align with the project's standards.
- Even though the new infix forms are functionally equivalent to the non-backtick ones, I believe it's important to test them as they introduce a new way of using the library. This ensures that the new syntax behaves as expected and helps maintain the overall quality of the codebase.

## Checklist
<!--- We'd like to thank you for your help, appreciate them and give you credit for it. Please check the checkboxes below as you complete them -->

- [x] I've added my name to the `AUTHORS` file, if it wasn't already present.

**Looking forward to your feedback!**


> Great project! If there's any way I can contribute further or assist with anything, please don't hesitate to let me know. I'm more than happy to collaborate and help out.

Commit:
Create fluent infixed backtick assertions

Commit:
Add name in AUTHORS

Commit:
Add examples in documentation for infix with objects backtick assertions

MarkusAmshove commented:
Hi, thank you for the contribution :-) I find the way you implemented this neat, especially making sure to not break any existing stuff :+1:

The builds currently break, because you're using backtick assertions in files like `ShouldBeAFileShould` and these methods don't work on Android.
Android currently still is in the `jvm` module, which is fine. To m ake sure the Android build works, there is a bit of [magic in the build files](https://github.com/MarkusAmshove/Kluent/blob/master/jvm/build.gradle#L32-L34) which skips files that contain `Backtick` in the name on Android :-)
To fix this, you'll have to take your tests out of the corresponding files and create new test files that have `Backtick` in their file name

Commit:
Fixed  test cases to be in its own file

Commit:
Renamed test cases to the previous versions

hvpaiva commented:
Hi, I have addressed the issue and made the necessary changes. I have moved the tests to new test files with "Backtick" in their file names to ensure compatibility with the Android build. Please let me know if there's anything else I need to do. Thanks! 😄

MarkusAmshove commented:
Awesome, thank you for the contribution 🥳 
I've merged the changes and will get to releasing them in a new version in the coming days, once I'm back at my machine where I've set up the deployment for maven central :-)

MarkusAmshove commented:
Just released 1.73 which includes these changes, should show up in maven central soon :)
